kernel/sockdev.c

```diff
@@ -660,6 +739,13 @@ static int _sd_close(myst_sockdev_t* sd, myst_sock_t* sock)
     if (!sd || !_valid_sock(sock))
         ERAISE(-EINVAL);
 
+    /* make sure it is shut down to make sure the socket gets unblocked */
+    if (sock->active_calls)
```

Comment:
There is a chance that `active_calls` is incremented immediately after this check returns false.

Reply:
active_calls is in fact atomic. And you are right, there is a small change it will still hang. This is why we will move forward with a more substantial fix #609 which turns the synchronous calls into synchronous ones under the covers